src/cq.rs

```diff
@@ -65,18 +123,17 @@ impl CompletionQueue {
         }
     }
 
+    pub fn borrow(&self) -> Result<CompletionQueueRef> {
+        self.handle.add_ref()?;
+        Ok(CompletionQueueRef { queue: self })
+    }
+
     /// Begin destruction of a completion queue.
     ///
     /// Once all possible events are drained then `next()` will start to produce
     /// `Event::QueueShutdown` events only.
     pub fn shutdown(&self) {
-        unsafe {
-            grpc_sys::grpc_completion_queue_shutdown(self.handle.cq);
-        }
-    }
-
-    pub fn as_ptr(&self) -> *mut GrpcCompletionQueue {
-        self.handle.cq
+        self.handle.unref()
```

Comment:
How about adding a flag so that CQ will not accept any new requests.

Reply:
`ref_cnt` ensures that already.